airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/migrations/AirbyteMessageMigrationV1_1_0.java

```diff
@@ -0,0 +1,241 @@
+package io.airbyte.commons.protocol.migrations;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.version.Version;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.v0.AirbyteStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.StreamSupport;
+
+public class AirbyteMessageMigrationV1_1_0 implements AirbyteMessageMigration<AirbyteMessage, io.airbyte.protocol.models.v0.AirbyteMessage> {
+
+  private static final Set<String> PRIMITIVE_TYPES = ImmutableSet.of(
+    "string",
+    "number",
+    "integer",
+    "boolean"
+  );
+  private static final Map<String, String> AIRBYTE_TYPE_TO_REFERENCE_TYPE = ImmutableMap.of(
+      "timestamp_with_timezone", "WellKnownTypes.json#definitions/TimestampWithTimezone",
+      "timestamp_without_timezone", "WellKnownTypes.json#definitions/TimestampWithoutTimezone",
+      "time_with_timezone", "WellKnownTypes.json#definitions/TimeWithTimezone",
+      "time_without_timezone", "WellKnownTypes.json#definitions/TimeWithoutTimezone",
+      "integer", "WellKnownTypes.json#definitions/Integer",
+      // these types never actually use airbyte_type, but including them for consistency
+      "string", "WellKnownTypes.json#definitions/String",
+      "number", "WellKnownTypes.json#definitions/Number",
+      "boolean", "WellKnownTypes.json#definitions/Boolean",
+      "date", "WellKnownTypes.json#definitions/Date"
+  );
+
+  @Override
+  public AirbyteMessage downgrade(io.airbyte.protocol.models.v0.AirbyteMessage oldMessage) {
+    return null;
```

Comment:
We should always return something in the downgrade since we'll be hitting this path for every message we send to a connector that is using protocol v0.

Reply:
:+1: I just forgot to put a todo in this method